pkg/build/api/types.go

```diff
@@ -76,8 +76,73 @@ type BuildSpec struct {
 	// scheduled in the system, that the build may be active on a node before the
 	// system actively tries to terminate the build; value must be positive integer
 	CompletionDeadlineSeconds *int64
+
+	// Info is a placeholder that can store multiple reasons why a build was
+	// triggered. It should cover for a future implementation of level triggered
+	// builds where you can multiple reasons why a build was triggered.
+	Info []*BuildTriggerInfo
```

Comment:
> My hold was partially because I may have changes to the design (because we need triggers across objects besides builds and deployments) and I don't want you to potentially waste time if we change direction fundamentally.

(moving @smarterclayton's comment to the right place....)

by "because we need triggers across objects besides builds and deployments" you mean "we need auditing/logging about triggers?  This feature isn't adding new triggers, it's just adding information about what trigger initiated the build.  (or what else initiated the build)